index.js

```diff
@@ -15,6 +15,7 @@ function getStream(inputStream, opts) {
 
 	var p = new Promise(function (resolve, reject) {
 		stream = bufferStream(opts);
+		inputStream.on('error', reject);
```

Comment:
> I believe the event propagates to piped streams.

Afaik it doesn't, which is one of the biggest pains with streams. Just look at [line #25](https://github.com/sindresorhus/get-stream/blob/90f4b75de963ad502457976c26ad9c14836fd2b5/index.js#L25), you are already listening for stream error, and it's not catching anything.